### PR TITLE
Parser 3.0: methods

### PR DESCRIPTION
**Pull request by hmdne**
Itself not related to the parser, but implements a few methods that were added between Ruby 2.6 and 3.0.

### CODE REVIEW

File: opal/corelib/comparable.rb [hunk 1]
```diff
@@ -80,7 +80,14 @@ def between?(min, max)
     true
   end
 
-  def clamp(min, max)
+  def clamp(min, max=nil)
+    if Range === min && max.nil?
+      max = min.end
+      max = Float::INFINITY if max.nil?
+      min = min.begin
+      min = -Float::INFINITY if min.nil?
```
Comment (elia) [inline opal/corelib/comparable.rb]: Can this be achieved with a simple `||`? I don't think we should consider `false` as a possible valid range edge

```rb
max = min.end || Float::INFINITY
min = min.being || -Float::INFINITY
```

File: opal/corelib/comparable.rb [hunk 2]
```diff
@@ -80,7 +80,7 @@ def between?(min, max)
     true
   end
 
-  def clamp(min, max=nil)
```
Comment (elia) [inline opal/corelib/comparable.rb]: Can you please squash the rubocop fixes to the initial commit in which the change was introduced?

File: opal/corelib/comparable.rb [hunk 3]
```diff
@@ -81,11 +81,15 @@ def between?(min, max)
...
-      min = -Float::INFINITY if min.nil?
+    if max.nil?
+      if Range === min
+        max = min.end
+        max = Float::INFINITY if max.nil?
+        min = min.begin
+        min = -Float::INFINITY if min.nil?
+      else
+        raise TypeError, "wrong argument type #{min.class} (expected Range)"
+      end
```
Comment (elia) [inline opal/corelib/comparable.rb:92]: Can you please add a quick explanation in the commit on why this was necessary?

**Review by elia — CHANGES_REQUESTED** [review]
Great work! ✨ 
Just a question, and the request to squash the rubocop fixes to the original commits.

File: opal/corelib/comparable.rb (hunk 1 above)
Comment (hmdne) [reply opal/corelib/comparable.rb]: false is a possible value and doesn't mean it's an infinite range

Comment (hmdne) [reply opal/corelib/comparable.rb]: i refactored this part anyway :)

**Comment by hmdne** [comment]
I will need to do a little bit of more work on this. But it seems good enough, of 700 fails added by a version change, this changeset removes 100 of them.

File: opal/corelib/comparable.rb (hunk 1 above)
Comment (elia) [reply opal/corelib/comparable.rb]: No, please if it's a possible range it's fine the way it is 👍 I just wasn't able to construct a range with a boolean, until now, it worked with `false..false`

**Comment by elia** [comment]
> I will need to do a little bit of more work on this. But it seems good enough, of 700 fails added by a version change, this changeset removes 100 of them.

@hmdne I'm always ok with splitting if you want to have the part of this that's already ok merged right away

**Comment by hmdne** [comment]
@elia If you see a problem to happen since some commit, we can split until that commit.

What do you think about the new approach to the #clamp? I basically ported this: https://github.com/ruby/ruby/blob/master/compar.c#L222-L252 (but to Javascript this time) and I found out that the entire module can be simplified.

**Comment by hmdne** [comment]
By the way, this force-push is only changed from f1c0c4f onwards (not counting the squashes).

**Review by elia — APPROVED** [review]
Great work! 
(I refined a bit the commits so that each fixed spec is enabled along with the fix)

**Pull request merged**